oci/auth/login/login.go

```diff
@@ -105,14 +108,57 @@ func (m *Manager) WithACRClient(c *azure.Client) *Manager {
 
 // Login performs authentication against a registry and returns the
 // authentication material. For generic registry provider, it is no-op.
-func (m *Manager) Login(ctx context.Context, url string, ref name.Reference, opts ProviderOptions) (authn.Authenticator, error) {
-	switch ImageRegistryProvider(url, ref) {
+func (m *Manager) Login(ctx context.Context, image string, ref name.Reference, opts ProviderOptions) (authn.Authenticator, error) {
+	regHost := splitRepositoryURL(image)
```

Comment:
are you talking about `splitRepositoryURL`?

Without doing this, we will pass full repo url to the azure login and it will make a call to the wrong url. (since we are no longer using registry host from ref.Context())

Reply:
We have come full circle😅. I did this at first, (only it was the same code in`ImageRegistryProvider()` that still uses the ref). I will make the change